Mathlib/Data/DFinsupp/Lex.lean

```diff
@@ -63,20 +63,22 @@ theorem lex_lt_of_lt [∀ i, PartialOrder (α i)] (r) [IsStrictOrder ι r] {x y
   exact lex_lt_of_lt_of_preorder r hlt
 #align dfinsupp.lex_lt_of_lt DFinsupp.lex_lt_of_lt
 
-instance Lex.isStrictOrder [LinearOrder ι] [∀ i, PartialOrder (α i)] :
+variable [LinearOrder ι]
+
+instance Lex.isStrictOrder [∀ i, PartialOrder (α i)] :
     IsStrictOrder (Lex (Π₀ i, α i)) (· < ·) :=
   let i : IsStrictOrder (Lex (∀ i, α i)) (· < ·) := Pi.Lex.isStrictOrder
   { irrefl := toLex.surjective.forall.2 fun _ ↦ @irrefl _ _ i.toIsIrrefl _
     trans := toLex.surjective.forall₃.2 fun _ _ _ ↦ @trans _ _ i.toIsTrans _ _ _ }
 #align dfinsupp.lex.is_strict_order DFinsupp.Lex.isStrictOrder
 
-variable [LinearOrder ι]
-
 /-- The partial order on `DFinsupp`s obtained by the lexicographic ordering.
 See `DFinsupp.Lex.linearOrder` for a proof that this partial order is in fact linear. -/
-instance Lex.partialOrder [∀ i, PartialOrder (α i)] : PartialOrder (Lex (Π₀ i, α i)) :=
-  PartialOrder.lift (fun x ↦ toLex (⇑(ofLex x)))
-    (FunLike.coe_injective (F := DFinsupp fun i => α i))
+instance Lex.partialOrder [∀ i, PartialOrder (α i)] : PartialOrder (Lex (Π₀ i, α i)) where
+  lt := (· < ·)
+  le x y := ⇑(ofLex x) = ⇑(ofLex y) ∨ x < y
```

Comment:
The `LE` on `Pi.Lex` is defined using [`partialOrderOfSO`](https://github.com/leanprover-community/mathlib4/blob/193761250a9869327056b499a7f49d676ed5e3a9/Mathlib/Order/RelClasses.lean#L189-L205), but if we use it here we would get `le x y := x = y ∨ x < y`, which is not defeq due to the absence of the coercion from DFinsupp to Function. I prefer the current defeq because it makes it possible to reuse the `Pi.Lex` results in `Lex.orderedAddCancelCommMonoid` etc., and because showing `⇑(ofLex x) = ⇑(ofLex y)` is not harder than showing `x = y`.